prog.go

```diff
@@ -35,14 +35,54 @@ const (
 // verifier log.
 const DefaultVerifierLogSize = 64 * 1024
 
+// LogLevel controls the verbosity of the kernel's eBPF program verifier.
+// These constants can be used for the ProgramOptions.LogLevel field.
+type LogLevel uint32
+
+const (
+	// Print verifier state at branch points.
+	LogLevelBranch LogLevel = iota + 1
+
+	// Print verifier state for every instruction.
+	// Available since Linux v5.2.
+	LogLevelInstruction
+
+	// Print verifier errors and stats at the end of the verification process.
+	// Available since Linux v5.2.
+	LogLevelStats
+
+	// Disable the verifier log.
+	LogLevelDisabled = math.MaxUint32
+)
+
 // ProgramOptions control loading a program into the kernel.
 type ProgramOptions struct {
-	// Controls the detail emitted by the kernel verifier. Set to non-zero
-	// to enable logging.
-	LogLevel uint32
-	// Controls the output buffer size for the verifier. Defaults to
-	// DefaultVerifierLogSize.
-	LogSize int
+	// Bitmap controlling the detail emitted by the kernel's eBPF verifier log.
+	// LogLevel-type values can be ORed together to request specific kinds of
+	// verifier output. See the documentation on [ebpf.LogLevel] for details.
+	//
+	//  opts.LogLevel = (ebpf.LogLevelBranch | ebpf.LogLevelStats)
+	//
+	// If left to its default value, the program will first be loaded without
+	// verifier output enabled. Upon error, the program load will be repeated
+	// with LogLevelBranch and the given (or default) LogSize value.
+	//
+	// Setting this to a value other than LogLevelDisabled will unconditionally
```

Comment:
Not true for 0?

Reply:
'Setting' implying something other than 0, since that's the default value. Clear IMO, but this might change anyway if we go for a bool to disable the log.